src/plugin-hooks/useSortBy.js

```diff
@@ -217,31 +217,38 @@ function useMain(instance) {
         rows,
         sortBy.map(sort => {
           // Support custom sorting methods for each column
-          const { sortType } = columns.find(d => d.id === sort.id)
-
-          // Look up sortBy functions in this order:
-          // column function
-          // column string lookup on user sortType
-          // column string lookup on built-in sortType
-          // default function
-          // default string lookup on user sortType
-          // default string lookup on built-in sortType
-          const sortMethod =
-            isFunction(sortType) ||
-            (userSortTypes || {})[sortType] ||
-            sortTypes[sortType] ||
-            sortTypes.alphanumeric
-
-          // Return the correct sortFn
-          return (a, b) =>
-            sortMethod(a.values[sort.id], b.values[sort.id], sort.desc)
+          const column = columns.find(d => d.id === sort.id)
+
+          if (column) {
+            const { sortType } = column
+            // Look up sortBy functions in this order:
+            // column function
+            // column string lookup on user sortType
+            // column string lookup on built-in sortType
+            // default function
+            // default string lookup on user sortType
+            // default string lookup on built-in sortType
+            const sortMethod =
+              isFunction(sortType) ||
+              (userSortTypes || {})[sortType] ||
+              sortTypes[sortType] ||
+              sortTypes.alphanumeric
+
+            // Return the correct sortFn
+            return (a, b) =>
+              sortMethod(a.values[sort.id], b.values[sort.id], sort.desc)
+          } else {
+            // if a column with id from sortBy state doesn't exist,
+            // return sorting function that keeps items in its original place
+            return () => 0
+          }
```

Comment:
Another way we could do this is filter the sortBy by column existence before the map. Then we don't need to fake anything at all. Thoughts?

Reply:
Sounds good, it's probably more performant for very large lists if there's a few of those columns that don't exist. Will take care of that.